core/controllers/skill_editor_test.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the skill editor page."""
+
+from core.domain import skill_services
+from core.domain import user_services
+from core.tests import test_utils
+import feconf
+
+
+class BaseSkillEditorControllerTest(test_utils.GenericTestBase):
+
+    def setUp(self):
+        """Completes the sign-up process for the various users."""
+        super(BaseSkillEditorControllerTest, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        self.set_admins([self.ADMIN_USERNAME])
+
+        self.admin = user_services.UserActionsInfo(self.admin_id)
+        self.skill_id = skill_services.get_new_skill_id()
+        self.save_new_skill(self.skill_id, self.admin_id, 'Description')
+
+
+class SkillEditorTest(BaseSkillEditorControllerTest):
+
+    def test_access_skill_editor_page(self):
+        """Test access to editor pages for the sample skill."""
+
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            # Check that non-admins cannot access the editor page.
+            response = self.testapp.get(
+                '%s/%s' % (feconf.SKILL_EDITOR_URL_PREFIX, self.skill_id))
+            self.assertEqual(response.status_int, 302)
+
+            # Check that admins can access and edit in the editor
+            # page.
+            self.login(self.ADMIN_EMAIL, is_super_admin=True)
```

Comment:
Ah, there are two concepts of admin in oppia. For access to skills use a user with ROLE_ADMIN rather than is_super_admin. Don't login with is_super_admin=True.

Ditto below.

Reply:
Done, it was because the check that I was using was can_access_admin_page which needed this. Changed the check to use a custom one that checks for admin.